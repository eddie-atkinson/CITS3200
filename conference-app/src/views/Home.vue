<template>
  <div>
    <v-container>
      <v-row
      align='center'
      justify='center'
      class='mt-4'
      >
        <v-card
        class='pa-3'
        >
          <v-card-title
          class='text-h4'
          >
            Conference Programme Creator
          </v-card-title>
          <v-card-text
          class='text-subtitle-1'
          >
           <p>
             Welcome to the Australian Institute for
             Geomechanics' conference programme creation app.
          </p>
           <p> To get started building your own conference programme click below!</p>
           <v-card-actions>
             <v-btn
             to='/build'
             >
               Get started
             </v-btn>
           </v-card-actions>
          </v-card-text>
        </v-card>
      </v-row>
    </v-container>
  </div>
</template>

<script>
// @ is an alias to /src

export default {
  name: 'Home',
  components: {
  },
  data() {
    return {
      stepNum: 1,
    };
  },
};
</script>
