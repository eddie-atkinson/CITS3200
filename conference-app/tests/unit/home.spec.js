import { shallowMount, createLocalVue } from '@vue/test-utils'; // Change this when vue init if required
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import Vue from 'vue';
import Home from '../../src/views/Home.vue'; // page being tested - Change this when vue init

Vue.use(Vuetify);

/*
None of the following will work until vue is appropriately installed and configured
REF https://jestjs.io/docs/en/getting-started for jest guide :)
*/

const localVue = createLocalVue();
localVue.use(VueRouter);

describe('Home View', () => {
  let vuetify;

  // Before each test we should ...
  beforeEach(() => {
    vuetify = new Vuetify(); // instantiate new Vuetify instance
  });

  test('is instantiated', () => {
    const wrapper = shallowMount(Home, {
      localVue,
      vuetify,
    }); // Mount our Home.vue for testing
    expect(wrapper.vm).toBeTruthy(); // Home component correctly instantiates
  });

  // CANT GET THIS TO WORK JUST YET :)

  /*
  it('has correct title', () => {
    let wrapper = shallowMount(Home, {
        localVue,
        vuetify,
      }); // Mount our Home.vue for testing
    const titleWrapper = wrapper.find('div');
    expect(titleWrapper.text()).toEqual('Conference Converter - Home');
  });
*/

  /*
    Snapshot testing, will initialise snapshot upon first run and all following
    If snapshot was intended to be modified, snapshot can be updated with npm test -- -u
    */
  it('matches last snapshot', () => {
    const wrapper = shallowMount(Home, {
      localVue,
      vuetify,
    }); // Mount our Home.vue for testing
    expect(wrapper.html()).toMatchSnapshot();
  });
});
