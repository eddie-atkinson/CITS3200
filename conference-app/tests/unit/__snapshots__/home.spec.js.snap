// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View matches last snapshot 1`] = `
<div>
  <v-container-stub tag="div">
    <v-row-stub tag="div" align="center" justify="center" class="mt-4">
      <v-card-stub loaderheight="4" tag="div" class="pa-3">
        <v-card-title-stub class="text-h4">
          Conference Programme Creator
        </v-card-title-stub>
        <v-card-text-stub class="text-subtitle-1">
          <p>
            Welcome to the Australian Institute for
            Geomechanics' conference programme creation app.
          </p>
          <p> To get started building your own conference programme click below!</p>
          <v-card-actions-stub>
            <v-btn-stub tag="button" activeclass="" to="/build" type="button" id="get-started-btn">
              Get started
            </v-btn-stub>
          </v-card-actions-stub>
        </v-card-text-stub>
      </v-card-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;
